### ISSUE TITLE
QModel Upgrade

### ISSUE THREAD

**Issue by PaulMacNichol** [QATCH-Technologies/nanovisQ#81]
# QModel Upgrade to Version 3 Checklist

The goal of this issue is to upgrade the QModel to Version 3 with the following enhancements and refactors:

## Feature Enhancements
- [ ] **QForecaster Integration**  
  - Incorporate `QForecaster` start-of-fill and end-of-fill suggestions into post-processing and training.
  - Incorporate `DoG_SVM_Score` feature and additional `QForecaster` features into the `QModel` feature set.
  - Add scaler and normalization from `QForecaster` to all features.

## Data and Training
- [x] **Training Data Update**  
  - Retrain on the new training data directory: `content/static/training`.
- [x] **Pre-training Optimization**  
  - Use objective function search pre-training.
- [x] **Tuning Method Update**  
  - Switch from using cross-validation in tuning to a balanced data approach.

## Post-Processing Improvements
- [x] **Simplify Post-Processing**  
  - Remove redundant/unnecessary post-processing steps.
- [ ] **Data Slicing Investigation**  
  - Investigate slicing the end of the dataset to adjust for multiplex data runs with long tails.

## Optimization and Hyperparameter Tuning
- [x] **Dynamic Hyperparameter Direction**  
  - Dynamically change the hyperparameter optimization direction to maximize for AUC or minimize for all other objectives.
- [x] **Optimization Library Review**  
  - Evaluate and potentially change the optimization library to Optuna for simplicity.

## Codebase Refactoring
- [x] **Component Separation**  
  - Break predictor and trainer components apart to prevent unnecessary code in the NanovisQ trunk.
- [x] **Model Consolidation Investigation**  
  - Investigate if it is possible to remove multiple `QMultiModel` types and consolidate to a single model without a clustering component.

**Comment by ajross4** [comment]
There are still 2 outstanding items to be addressed in future development work. Tagging SW v2.6b57 from the trunk today with other changes related to this issue that introduce `QModel v3` to the nanovisQ software ecosystem.
